experimental/unicodeset_parser/src/parse.rs

```diff
@@ -407,7 +384,7 @@ where
                 }
                 // parse a literal char (either individually or as the start of a range)
                 (Begin | Char | AfterUnicodeSet, c) if legal_char_start(c) => {
-                    let c = self.parse_char(self.options.dollar_is_anchor)?;
+                    let c = self.parse_char(true)?;
```

Comment:
Nit for future: avoid functions that take boolean arguments because reading this code I have no idea what `true` is supposed to mean in this context without looking at the signature of `parse_char`